components/browser/menu/src/main/java/mozilla/components/browser/menu/item/PlaceholderBrowserMenuItem.kt

```diff
@@ -0,0 +1,30 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.menu.item
+
+import android.view.View
+import android.widget.TextView
+import mozilla.components.browser.menu.BrowserMenu
+import mozilla.components.browser.menu.BrowserMenuItem
+import mozilla.components.browser.menu.R
+
+/**
+ * A browser menu item that is to be used only as a placeholder for inserting web extensions in main menu.
+ * The id of the web extension to be inserted has to correspond to the label of the browser menu item.
+ *
+ * @param label The label of this menu item.
+ */
+class PlaceholderBrowserMenuItem(
```

Comment:
Since this is specifically for web extensions, should we reflect that in the name e.g. `WebExtensionPlaceholderMenuItem` or similar / shorter?

Reply:
I'm a bit confused about if we really need the `PlaceholderBrowserMenuItem` class, as it's not a real UI component, as mentioned in the docs, its only purpose is for inserting web extensions in the main menu, and we are using it for [holding the id](https://github.com/mozilla-mobile/fenix/pull/15884/files#diff-74ecbb7fa46183fe3df86ce45428e0a4b2d7d4039d2989efea385b0baac19910R289) of a web extension and finding the `WebExtensionBrowserMenuItem` equivalent, could it be easier to pass a list of ids of promoted extensions without creating an extra class? I think this way we have a more intentional APi and less code to maintain. What do you think? :)

Reply:
I actually like the current approach, because it makes it easy to read / explicit in the menu builder: https://github.com/mozilla-mobile/fenix/pull/15884/files#diff-74ecbb7fa46183fe3df86ce45428e0a4b2d7d4039d2989efea385b0baac19910R193

I can clearly see where the menu item will end up and **at which position**.

Reply:
Indicating the item position is good plus 👍🏽

Reply:
Fine either way :). I do like how this API solves the problem of where to put the menu item without having to specify both the extension ID and the index.

Reply:
Without a placeholder, it would be difficult to insert the extension in both top toolbar and bottom toolbar items order.

Reply:
👍🏽